src/frontend/src/eth/components/fee/FeeContext.svelte

```diff
@@ -97,7 +98,13 @@
 				...(await getFeeData()),
 				gas: await getErc20FeeData({
 					...erc20GasFeeParams,
-					targetNetwork
+					targetNetwork,
+					to:
+						// When converting "ICP Erc20" to native ICP, the destination address is an "old" ICP hex account identifier.
+						// Therefore, it should not be prefixed with 0x.
+						nonNullish(targetNetwork) && isNetworkICP(targetNetwork)
```

Comment:
just as side note, we could actually have that `isNetworkXXX` accepts `undefined` at some point, no?

Reply:
Definitely! Can you do that afterwards?

Reply:
Done:

- PR #2057